src/app/unlocked-stage/unlocked-stage.component.ts

```diff
@@ -1,156 +1,331 @@
+    this.$mosquito.attr({'class': 'moz'});
+    this.$mosquito.append(progress);
 
-    stopAttack() {
-        this._spray.src = '../assets/img/unlocked-stage/spray-cursor.gif';
-        this._mousedown = false;
+    // Adding n = mosCountInitial initially in the game
+    for (let i = 1; i <= this.mosCountInitial; i++) {
+      const mosquitoClone = this.$mosquito.clone();
+      $('#scene').append(mosquitoClone);
+      this.mosquitoes.push(mosquitoClone);
     }
 
-    moveOverCanvas(e) {
-        this._canvas.style.cursor = 'none';
-        this._spray.style.display = 'block';
-        this._spray.style.left = e.pageX - 425 + 'px';
-        this._spray.style.top = e.pageY - 150 + 'px';
+    // Spray element
+    this.$spray = $('<img src="../assets/img/unlocked-stage/spray-cursor.gif">');
+    this.$spray.attr({'id': 'spray'});
+    $('#scene').append(this.$spray);
+
+    // Net element
+    this.$net = $('<img src="../assets/img/unlocked-stage/net.png">');
+    this.$net.attr({'id': 'net'});
+    $('#scene').append(this.$net);
+    this.$net.css({display: 'block'});
+    this.$net.css({left: '840px'});
+
+    // Progress and HealthBars
+    this._humanProgBar = document.getElementById('human-health-bar');
+
+    // Human
+    this.$human = $('<img src="../assets/img/unlocked-stage/human.png">').attr({'id': 'human'});
+    $('#scene').append(this.$human);
+
+    // Time increment begins as the game starts
+    this.incrementTime();
+    this._spray = this.$spray;
+    this._mousemoveListener = this._renderer.listen(this._canvas, 'mousemove', (event) => this.moveOverCanvasSpray(event));
+    this._mouseleaveListener = this._renderer.listen(this._canvas, 'mouseleave', (event) => this.leaveCanvas());
+    this._mousedownListener = this._renderer.listen(this._canvas, 'mousedown', (event) => this.attack());
+    this._mouseupListener = this._renderer.listen(this._canvas, 'mouseup', (event) => this.stopAttack());
+    const that = this;
+    this.$net.mouseenter(function () {
+      that._renderer.listen(that._canvas, 'mousemove', (event) => that.moveOverCanvasNet(event))
+    });
+  }
+
+  measureSpray() {
+    const that = this;
+    setTimeout(function () {
+      if (that._mousedown) {
+        that.stopAttack();
+        that.sprayValue -= 4;
+        that.measureSpray();
+      }
+    }, 500)
+  }
+
+  attack() {
+    this.$spray.attr('src', '../assets/img/unlocked-stage/spray.gif');
+    this._mousedown = true;
+    this.measureSpray();
+  }
+
+  stopAttack() {
+    this.$spray.attr('src', '../assets/img/unlocked-stage/spray-cursor.gif');
+    this._mousedown = false;
+  }
+
+  moveOverCanvasSpray(e) {
+    this._canvas.style.cursor = 'none';
+    this._spray.css({display: 'block'});
+    this._spray.css({left: e.pageX - 510 + 'px'});
+    this._spray.css({top: e.pageY - 150 + 'px'});
+  }
+
+  moveOverCanvasNet(e) {
+    if (!(e.pageX > 1270 || e.pageX < 1200 || e.pageY < 517)) {
```

Comment:
Can constants be defined for all these values?

Reply:
Done.